### PR TITLE
doc: improved README.md testing info

### PR DESCRIPTION
**Pull request by IuliaBlindu**
## Checklist

- [x] PR name added as appropriate (e.g. `feat:`/`fix:`/`doc:`/`test:`/`refactor:`)
- [ ] Tests added or updated
- [ ] Documentation updated
- [ ] Changelog updated and version incremented (label: upcoming release)
- [ ] Github Issue linked if any
- [x] Jira task updated

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -67,23 +67,39 @@ $ $GOPATH/bin/terraform-provider-ionoscloud
...
 
-⚠️ **_Note: Acceptance tests create real resources, and often cost money to run._**
+### Test Tags
 
-```sh
-$ make testacc TAGS=all
+If you want to test a specific batch of tests, they are split by **build tags** as follows:
+
+- `compute` - all **compute engine** tests (datacenter, firewall rule, image, IP block, IP failover, lan, location, nic, private cross connect, server, snapshot, template, volume)
+- `nlb` - **network load balancer network load balancer forwarding rule tests**
```
Comment (mihaela-mh) [inline README.md:91]: it seems like *network load balancer* is duplicated here, maybe there is a missing **and**?

File: README.md [hunk 2]
```diff
@@ -67,23 +67,39 @@ $ $GOPATH/bin/terraform-provider-ionoscloud
...
+- `natgateway` - **NAT gateway and NAT gateway rule tests**
+- `k8s` - **k8s cluster and k8s node pool tests**
+- `dbaas` - **postgres cluster tests**
```
Comment (mihaela-mh) [inline README.md:94]: **postgres cluster tests** may become **DBaaS postgres cluster tests** (just for clarity)

File: README.md [hunk 3]
```diff
@@ -67,23 +67,39 @@ $ $GOPATH/bin/terraform-provider-ionoscloud
 ...
 ```
 
+## Testing the Provider
+
+⚠️ **Warning:** Acceptance tests create real resources, and often cost money to run.
```
Comment (AnaMariaMoga) [inline README.md]: In terraform, there are only acceptance tests? I had the impression we have e2e tests.

File: README.md [hunk 4]
```diff
@@ -67,23 +67,39 @@ $ $GOPATH/bin/terraform-provider-ionoscloud
...
+``` bash
+$ make testacc TAGS=dbaas
 ```
 
-Will run all server and lan tests:
-```sh
-$ make testacc TAGS=server,lan
-```
\ No newline at end of file
+You can also test one single resource, using on of the tags: `backup`, `datacenter`, `dbaas`, `firewall`, `group`, `image`, `ipblock`, `ipfailover`, `k8s`, `lan`, `location`, `natgateway`,
```
Comment (mihaela-mh) [inline README.md:100]: using **one** of the tags
^ there's a typo here

File: README.md [hunk 5]
```diff
@@ -67,23 +67,39 @@ $ $GOPATH/bin/terraform-provider-ionoscloud
...
+`nlb`, `nic`, `pcc`, `resource`, `s3key`, `server`, `share`, `snapshot`, `template`, `user`, `volume`
+
+### What Are We Testing?
+
+The purpose of our tests is to **provision** resources containing all the available arguments, followed by **updates** on all arguments that allow this action. Beside the provisioning part, **data-sources** with all possible arguments and **imports** are also tested.
```
Comment (AnaMariaMoga) [inline README.md:105]: I would add details about the daily runs. Something like: tests are integrated into a github action and the CI workflows can be found here or tests are run daily and you can check the status here.

File: README.md [hunk 6]
```diff
@@ -67,23 +67,39 @@ $ $GOPATH/bin/terraform-provider-ionoscloud
...
 
 In order to run the full suite of Acceptance tests, run `make testacc TAGS=all`.
 
-Tests can be split using build tags: all(all tests), compute(all compute-engine tests), dbaas, k8s, server, lan, ipblock.
+``` bash
+$ make testacc TAGS=all
+```
 
-⚠️ **_Note: Acceptance tests create real resources, and often cost money to run._**
+### Test Tags
```
Comment (AnaMariaMoga) [inline README.md:86]: From my perspective, there is a **lot** of information about testing. At a first look, it seemed to me we have information about testing for 1/2 of the README.md file. Personally, I would wrap some of the info with a toggle button. 

For me, the main point with testing info in a README is to be _transparent_ about what tests (unit, e2e, acceptance) are present, how to run them locally (with one note about the tags) and CIs.

File: README.md (hunk 3 above)
Comment (IuliaBlindu) [reply README.md]: In this case acceptance tests involves e2e tests, since they create real resources

**Review by AnaMariaMoga — APPROVED** [review]

**Review by cristiGuranIonos — APPROVED** [review]

**Pull request merged**